source/ota.c

```diff
@@ -1262,45 +1155,34 @@ static DocParseErr_t decodeAndStoreKey( char * pValueInJson,
                                         void * pParamAdd )
 {
     DocParseErr_t err = DocParseErrNone;
+    size_t actualLen = 0;
+    Base64Status_t base64Status = 0;
+    Sig256_t ** pSig256 = pParamAdd;
 
-    /* Allocate space for and decode the base64 signature. */
-    void * pSignature = otaAgent.pOtaInterface->os.mem.malloc( sizeof( Sig256_t ) );
+    /* pSig256 should point to pSignature in OtaFileContext_t, which is statically allocated. */
+    assert( *pSig256 != NULL );
```

Comment:
Is there an upper bound for the size of a signature that we can receive in a job document? I didn't think that there was.

Reply:
See `kOTA_MaxSignatureSize`, it's part of `Sig256_t` struct.